### PR TITLE
Major code cleanup for building on Linux

### PR DESCRIPTION
**Pull request by dascandy**
No changes that will affect Windows development, but this gets you free portability. Looks like a big change request, but 95%+ is just changing backslashes to forward slashes.

### CODE REVIEW

File: include/Managers/AnimationManager.h [hunk 1]
```diff
@@ -29,5 +29,5 @@ class AnimationManager
 private:
 	std::unordered_map<std::string, Animation> m_animations;
 	Animation* m_currentAnimation;
-	SharedContext m_sharedContext;
```
Comment (dascandy) [inline include/Managers/AnimationManager.h:32]: SharedContext is not copyable. This should not have compiled.

Comment (dascandy) [reply include/Managers/AnimationManager.h:32]: ... This class is now excluded from build. Maybe remove it entirely?

**Comment by dascandy** [comment]
This now results in a build on Linux targets.

**Comment by 0x2648** [comment]
Works for me! Under Windows 7 64bit with MinGW 5.3.0 32bit i had to add 

        struct EnumClassHash
        {
            template <typename T>
            std::size_t operator()(T t) const
            {
                return static_cast<std::size_t>(t);
            }
        };

for the *unordered_map* in *EventManager* and *StateManager* (see [here](http://stackoverflow.com/questions/18837857/cant-use-enum-class-as-unordered-map-key)).

I would also suggest to add 

    set(CMAKE_CXX_STANDARD 14)

to the CMakeLists.txt and rewrite the link against SFML (see [here](http://blog.beuc.net/posts/Build_a_SFML_project_with_CMake/)).

**Comment by dascandy** [comment]
The link against SFML should use a find_package; I was a bit lazy in getting it to work. I'll fix the latter two & send another pull request after this one is accepted by @RyanSwann1 ; will you pick up the hash @0x2648 ?

**Comment by dascandy** [comment]
I've pushed the code cleanup (remove dead source files, remove commented-out code) into a separate branch. Let me know if you want it or not - many of those bits could actually be pending integration rather than old code.

**Comment by 0x2648** [comment]
@dascandy sure, i will pick up the hash.

**Pull request closed without merging**